File: frontend/src/types/assignment.ts
export type AssignmentSubmissionStatus = 'submitted' | 'reviewed' | 'returned';

export interface AssignmentSubmission {
  id: number;
  lesson_id: number;
  course_id: number;
  student_id: number;
  text: string | null;
  file_name: string | null;
  file_size: number | null;
  status: AssignmentSubmissionStatus;
  score: number | null;
  feedback: string | null;
  submitted_at: string;
  reviewed_at: string | null;
  reviewed_by_id: number | null;
}

File: backend/app/schemas/assignment.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.assignment import AssignmentSubmissionStatus


class AssignmentSubmissionOut(BaseModel):
    id: int
    lesson_id: int
    course_id: int
    student_id: int
    text: str | None
    file_name: str | None
    file_size: int | None
    status: AssignmentSubmissionStatus
    score: int | None
    feedback: str | None
    submitted_at: datetime
    reviewed_at: datetime | None
    reviewed_by_id: int | None

    model_config = {"from_attributes": True}


class AssignmentReviewIn(BaseModel):
    status: AssignmentSubmissionStatus = AssignmentSubmissionStatus.reviewed
    score: int | None = Field(default=None, ge=0, le=100)
    feedback: str | None = None
